kiota_abstractions/request_headers.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+from typing import Dict, List, Set, Union
+
+
+class RequestHeaders():
+    "Represents a collection of request headers"
+
+    def __init__(self) -> None:
+        self._headers: Dict[str, Set[str]] = {}
+        self._single_value_headers: Set[str] = {
```

Comment:
could this be static instead? we don't need an instance of those per headers class instance.

Reply:
I'm not sure I follow. I assumed we create `HeadersCollection` instances and track changes per instance. No?

Reply:
I was referring to the collection of single valued headers.